raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -936,13 +937,13 @@ private CompletableFuture<FetchResponseData> handleFetchRequest(
         RaftRequest.Inbound requestMetadata,
         long currentTimeMs
     ) {
-        FetchRequestData request = (FetchRequestData) requestMetadata.data;
+        FetchRequest request = new FetchRequest((FetchRequestData) requestMetadata.data, requestMetadata.apiVersion);
```

Comment:
I am not sure to understand why we need this change. Could you elaborate?

Reply:
The FetchRequestData can have versions 14 and 15. The main problem is that KafakRaftClient needs the ReplicaId which is not at the field of FetchRequestData. We need the API version to properly locate the ReplicaId.

Reply:
reverted